contracts/chain-adapters/l2/WithdrawalAdapter.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.19;
+
+import { CircleCCTPAdapter, ITokenMessenger, CircleDomainIds } from "../../libraries/CircleCCTPAdapter.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @title Adapter for interacting with bridges from a generic L2 to Ethereum mainnet.
+ * @notice This contract is used to share L2-L1 bridging logic with other Across contracts.
+ */
+abstract contract WithdrawalAdapter is CircleCCTPAdapter {
```

Comment:
Prefer naming this with Base suffix to be consistent

Reply:
That is better https://github.com/across-protocol/contracts/pull/605/commits/4db2572d1a4b4bf7ebb81d87659f57e3afda4f0b